src/f32ext/ln.rs

```diff
@@ -0,0 +1,113 @@
+use super::abs;
+use super::utils;
+use super::utils::FloatComponents;
+/// Floating natural log
+use core::f32;
+use core::u32;
+
+//excessive precision doesn't matter that much.
+#[allow(clippy::excessive_precision)]
+pub(super) fn ln_1to2_series_approximation(x: f32) -> f32 {
+    // idea from https://stackoverflow.com/a/44232045/
+    // modified to not be restricted to int range and only values of x above 1.0.
+    // and got rid of most of the slow conversions,
+    // should work for all positive values of x.
+    // log base 2(E) == 1/ln(2)
+    //let recip_ln2: f32 = f32::consts::LOG2_E;
+
+    //clippy forces me to do this.
```

Comment:
It'd be more helpful for the comment here to note clippy's rationale for this

Reply:
I wasn't used to the rational, it was basically because you can't guarantee direct comparison of floats that would realistically be close enough will be exactly the same number.  I'll go over the code base again to clean up some of these issues when I get a chance today.